tfexec/apply.go

```diff
@@ -99,6 +100,38 @@ func (tf *Terraform) Apply(ctx context.Context, opts ...ApplyOption) error {
 	return tf.runTerraformCmd(ctx, cmd)
 }
 
+// ApplyJSON represents the Terraform apply subcommand with the `-json` flag.
+// Using the `-json` flag will result in
+// [machine-readable](https://developer.hashicorp.com/terraform/internals/machine-readable-ui)
+// JSON being written to the supplied `io.Writer`.
+func (tf *Terraform) ApplyJSON(ctx context.Context, w io.Writer, opts ...ApplyOption) error {
+	err := tf.compatible(ctx, tf0_15_3, nil)
+	if err != nil {
+		return fmt.Errorf("terraform apply -json was added in 0.15.3: %w", err)
+	}
+
+	tf.SetStdout(w)
+
+	cmd, err := tf.applyJSONCmd(ctx, opts...)
+	if err != nil {
+		return err
+	}
+
+	return tf.runTerraformCmd(ctx, cmd)
+}
+
+func (tf *Terraform) applyJSONCmd(ctx context.Context, opts ...ApplyOption) (*exec.Cmd, error) {
+	cmd, err := tf.applyCmd(ctx, opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	cmd.Args = append(cmd.Args[:3], cmd.Args[2:]...)
+	cmd.Args[2] = "-json"
```

Comment:
I'm assuming the slice contains 2 arguments for sure `[]string{"terraform", "apply", ...}` and while it's _likely_ there will be other arguments after `apply`, I'm not sure it's a good idea to rely on it as a change in the underlying `applyCmd()` could cause panic here.

TL;DR Is there a strong reason for replacing the argument, instead of just appending `-json` to the end?

Reply:
Refactored to split the building of the args into flags that preceed the optional dir/plan which must come after all flags.